hawkauthlib/__init__.py

```diff
@@ -207,23 +213,21 @@ def check_signature(request, key, hashmod=None, params=None, nonces=None):
         params = utils.parse_authz_header(request, {})
     if params.get("scheme") != "Hawk":
         return False
+    if require_payload_hash and "hash" not in params:
+        return False
     # Any KeyError here indicates a missing parameter.
     # Any ValueError here indicates an invalid parameter.
     try:
         timestamp = int(params["ts"])
         nonce = params["nonce"]
-        # Check validity of the signature, checking the MAC first is faster
-        # https://github.com/mozilla/hawk/blob/main/API.md
-        expected_sig = get_signature(request, key, hashmod, params, accept_untrusted_content=True)
-        if utils.strings_differ(params["mac"], expected_sig):
-            return False
-        # If the MAC is valid, the server calculates the payload hash and compares the value with
-        # the provided payload hash in the header
-        # https://github.com/mozilla/hawk/blob/main/API.md
-        expected_sig = get_signature(request, key, hashmod, params, accept_untrusted_content=False)
```

Comment:
Calling `get_signature` again here seems like unnecessary overhead and we should just compare against the `hash` parameter which we already know is validly included in the signature. (I believe this is what the spec means by "If the MAC is valid, the server calculates the payload hash and compares the value with the provided payload hash in the header").

Reply:
The comment links to the API which states there is a fast check and if that fast check is True you then do the slow but secure check

Reply:
Also, I would prefer to do what mohawk did, and skip the fast check but for historical reasons hawkhashlib _only_ did the fast (insecure) check.. so do we simply decide now that we only do the secure check and never allow insecure / untrusted hash checks?